pattern-hues.lic

```diff
@@ -47,23 +46,34 @@ class PatternHues
   end
 
   def cast_pattern_hues
-    match_messages = [
+    match_successes = [
       /You trace the complex sigils of the Pattern Hues cantrip around yourself./,
       /You already have/,
       /You gesture/,
+      /You are now prepared to cast the Pattern Hues cantrip./,
+
+    ]
+    match_failures = [
       /Something in the area is interfering/,
       /You try, but the dolphin choose/,
       /You should stop practicing/,
-      /You find it hard to concentrate on anything/
+      /You find it hard to concentrate on anything/,
+      /you suddenly realize that there is no custom color/,
+      /You'll interrupt your research!/
     ]
-    scripts_to_unpause = DRC.smart_pause_all
-    bput("prep cantrip pattern hues", "You are now prepared to cast the Pattern Hues cantrip.","You already have", "You should stop practicing")
+
+    @scripts_to_unpause = DRC.safe_pause_list
+    unless @scripts_to_unpause
+      DRC.message("Cannot pause, trying again shortly.")
+      return
+    end
+    DRC.bput("prep cantrip pattern hues", *match_successes, *match_failures)
```

Comment:
So this prevents a hang, but it bulls through to the gesture, even if it hits failure. I've tried this:
```ruby
unless /You are now prepared/i =~ DRC.bput("prep cantrip pattern hues", *match_successes, *match_failures)
  DRC.safe_unpause_list(scripts_to_unpause)
  return
end```

Which, as I understand it, accepts only a successful prep, otherwise, unpauses, then returns to wait the 600 second duration before another attempt.

Reply:
I can review this tonight, maybe.

Reply:
Since the pattern for successes could contain more than one, I changed to a case statement:
```ruby
case DRC.bput("prep cantrip pattern hues", *match_successes, *match_failures)
    when *match_failures
        if "That is not a cantrip you can cast"
          exit
        else
          DRC.safe_unpause_list(scripts_to_unpause)
          return
        end
    end
    ```
    Tested on one char who does not have pattern hues yet. Will test on Dartellum after his hunt.

Reply:
@Raykyn55 please try the update and let me know.